File: api/app/Models/Invoice.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class Invoice extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'order_id', 'invoice_series_id', 'doc_type', 'serie', 'correlativo', 'full_number',
        'status', 'customer_doc_type', 'customer_doc_number', 'customer_name',
        'currency', 'form_of_payment', 'payment_method_id',
        'mto_oper_gravadas', 'mto_igv', 'valor_venta', 'sub_total', 'mto_imp_venta',
        'sunat_code', 'sunat_description', 'sunat_notes', 'xml_content', 'cdr_content',
        'note_motive', 'note_motive_desc', 'ref_doc_type', 'ref_doc_number', 'ref_doc_date',
        'observations', 'issued_at', 'user_id',
    ];

    protected $casts = [
        'mto_oper_gravadas' => 'decimal:2',
        'mto_igv'           => 'decimal:2',
        'valor_venta'       => 'decimal:2',
        'sub_total'         => 'decimal:2',
        'mto_imp_venta'     => 'decimal:2',
        'sunat_notes'       => 'array',
        'issued_at'         => 'datetime',
        'ref_doc_date'      => 'date',
    ];

    /** Nombre legible del tipo de comprobante */
    public function getDocTypeLabelAttribute(): string
    {
        return match ($this->doc_type) {
            '01' => 'Factura',
            '03' => 'Boleta de Venta',
            '07' => 'Nota de Crédito (Factura)',
            '08' => 'Nota de Crédito (Boleta)',
            default => 'Comprobante',
        };
    }

    /** Nombre legible del estado */
    public function getStatusLabelAttribute(): string
    {
        return match ($this->status) {
            'draft'     => 'Borrador',
            'pending'   => 'Enviando',
            'accepted'  => 'Aceptado',
            'rejected'  => 'Rechazado',
            'exception' => 'Excepción',
            'error'     => 'Error',
            'cancelled' => 'Anulado',
            default     => ucfirst($this->status),
        };
    }

    public function order(): BelongsTo
    {
        return $this->belongsTo(Order::class);
    }

    public function paymentMethod(): BelongsTo
    {
        return $this->belongsTo(PaymentMethod::class);
    }

    public function invoiceSeries(): BelongsTo
    {
        return $this->belongsTo(InvoiceSeries::class, 'invoice_series_id');
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Sales/Entities/Invoice.cs
using AdminHiitop.Api.Domain.Catalog.Entities;
using AdminHiitop.Api.Domain.Common;

namespace AdminHiitop.Api.Domain.Sales.Entities;

public sealed class Invoice : AuditableEntity
{
    public int? OrderId { get; set; }
    public int InvoiceSeriesId { get; set; }
    public string DocType { get; set; } = string.Empty;
    public string Serie { get; set; } = string.Empty;
    public int Correlativo { get; set; }
    public string FullNumber { get; set; } = string.Empty;
    public string Status { get; set; } = "draft";
    public string? CustomerDocType { get; set; }
    public string? CustomerDocNumber { get; set; }
    public string? CustomerName { get; set; }
    public string Currency { get; set; } = "PEN";
    public string FormOfPayment { get; set; } = "contado";
    public int? PaymentMethodId { get; set; }
    public decimal MtoOperGravadas { get; set; }
    public decimal MtoIgv { get; set; }
    public decimal ValorVenta { get; set; }
    public decimal SubTotal { get; set; }
    public decimal MtoImpVenta { get; set; }
    public int? SunatCode { get; set; }
    public string? SunatDescription { get; set; }
    public string? SunatNotesJson { get; set; }
    public string? XmlContent { get; set; }
    public string? CdrContent { get; set; }
    public string? NoteMotive { get; set; }
    public string? NoteMotiveDesc { get; set; }
    public string? RefDocType { get; set; }
    public string? RefDocNumber { get; set; }
    public DateTime? RefDocDate { get; set; }
    public string? Observations { get; set; }
    public DateTime IssuedAt { get; set; }
    public int? UserId { get; set; }

    public Order? Order { get; set; }
    public InvoiceSeries InvoiceSeries { get; set; } = null!;
    public PaymentMethod? PaymentMethod { get; set; }
    public User? User { get; set; }
    public ICollection<DailySummaryItem> DailySummaryItems { get; set; } = new List<DailySummaryItem>();
    public ICollection<SunatSendLog> SunatSendLogs { get; set; } = new List<SunatSendLog>();
}
